src/plots/cartesian/axes.js

```diff
@@ -706,11 +707,16 @@ axes.calcTicks = function calcTicks(ax, opts) {
                 ) ax.dtick = 'M12';
             }
         }
+
+        if(prevDtick !== ax.dtick) {
+            // move tick0 back
+            ax.tick0 = axes.tickIncrement(ax.tick0, prevDtick, !axrev, ax.calendar);
```

Comment:
This seems problematic - if you set a specific `tick0` it won't be honored? In any event I don't think we can be pushing a different value back into `ax` at this point, but if `tickFirst` needs a different effective start point perhaps we could make it an option to `tickFirst`?

Reply:
Good call. Addressed in a4cb02b.